public/app/features/dashboard/components/Inspector/PanelInspector.tsx

```diff
@@ -1,42 +1,187 @@
-// Libraries
 import React, { PureComponent } from 'react';
 
 import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
-import { Drawer, JSONFormatter } from '@grafana/ui';
-import { css } from 'emotion';
-import { getLocationSrv } from '@grafana/runtime';
+import { JSONFormatter, Drawer, Select, Table } from '@grafana/ui';
+import { getLocationSrv, getDataSourceSrv } from '@grafana/runtime';
+import { DataFrame, DataSourceApi, SelectableValue, applyFieldOverrides } from '@grafana/data';
+import { config } from 'app/core/config';
 
 interface Props {
   dashboard: DashboardModel;
   panel: PanelModel;
 }
 
-export class PanelInspector extends PureComponent<Props> {
+enum InspectTab {
+  Data = 'data',
+  Raw = 'raw',
+  Issue = 'issue',
+  Meta = 'meta', // When result metadata exists
+}
+
+interface State {
+  // The last raw response
+  last?: any;
+
+  // Data frem the last response
+  data: DataFrame[];
+
+  // The selected data frame
+  selected: number;
+
+  // The Selected Tab
+  tab: InspectTab;
+
+  // If the datasource supports custom metadata
+  metaDS?: DataSourceApi;
+}
+
+export class PanelInspector extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      data: [],
+      selected: 0,
+      tab: InspectTab.Data,
+    };
+  }
+
+  async componentDidMount() {
+    const { panel } = this.props;
+    if (!panel) {
+      this.onDismiss(); // Try to close the component
+      return;
+    }
+
+    // TODO? should we get the result with an observable once?
+    const lastResult = (panel.getQueryRunner() as any).lastResult;
+    if (!lastResult) {
+      this.onDismiss(); // Usually opened from refresh?
+      return;
+    }
+
+    // Find the first DataSource wanting to show custom metadata
+    let metaDS: DataSourceApi;
+    const data = lastResult?.series as DataFrame[];
+    if (data) {
+      for (const frame of data) {
+        const key = frame.meta?.ds?.dataSourceName;
+        if (key) {
+          const ds = await getDataSourceSrv().get(key);
+          if (ds && ds.components.MetadataInspector) {
+            metaDS = ds;
+            break;
+          }
+        }
+      }
+    }
+
+    // Set last result, but no metadata inspector
+    this.setState({
+      last: lastResult,
+      data,
+      metaDS,
+    });
+  }
+
   onDismiss = () => {
     getLocationSrv().update({
       query: { inspect: null },
       partial: true,
     });
   };
 
+  onSelectTab = (item: SelectableValue<InspectTab>) => {
+    this.setState({ tab: item.value || InspectTab.Data });
+  };
+
+  onSelectedFrameChanged = (item: SelectableValue<number>) => {
+    this.setState({ selected: item.value || 0 });
+  };
+
+  renderMetaDataInspector() {
```

Comment:
in other places we write it Metadata (lower case D), maybe it should be MetaDataInspector ? or lowercase d is correct.